### PR TITLE
feat: Use Selector, cleanup, and return success

### PR DESCRIPTION
**Pull request by deluca-mike**
Fixed tests, too.

### CODE REVIEW

**Review by satyamakgec — COMMENTED** [review]
I wasn't aware of this helper repo.  Anyway, my question is do we really need this low-level transfer handling instead of SafeERC20 lib for transfers.

File: src/ERC20Helper.sol [hunk 1]
```diff
@@ -1,42 +1,32 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity ^0.8.7;
 
-//@title Small Library to standardize erc20 token interactions
+import { IERC20 } from "../lib/erc20/src/interfaces/IERC20.sol";
+
+///@title Small Library to standardize erc20 token interactions
```
Comment (lucas-manuel) [inline src/ERC20Helper.sol:6]: `/// @title`

File: src/ERC20Helper.sol [hunk 2]
```diff
@@ -1,42 +1,32 @@
...
-    function _call(address token, bytes memory data) private {
-        (bool ok, bytes memory returnData) = token.call(data);
+    function approve(address token, address spender, uint256 amount) internal returns (bool) {
+        return _call(token, abi.encodeWithSelector(IERC20.approve.selector, spender, amount));
+    }
 
-        require(ok);
+    function _call(address token, bytes memory data) private returns (bool success) {
+        bytes memory returnData;
+        (success, returnData) = token.call(data);
```
Comment (lucas-manuel) [inline src/ERC20Helper.sol:27]: I think we can do `(success, bytes memory returnData)` if I'm not mistaken

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Very nice - just small changes, also we need a CI file

File: src/ERC20Helper.sol (hunk 2 above)
Comment (deluca-mike) [reply src/ERC20Helper.sol:27]: Can't. Either they are all declared in that line, or none of them are.

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**